classes/hooks/synchronization/Segment.php

```diff
@@ -173,12 +173,13 @@ private function _create($res_contacts, $filterId, $fiterName)
 			try {
                 
                  
-                $file = new SplFileObject('contacts.csv', 'w');
+                $file = fopen('contacts.csv', 'w');
```

Comment:
Instead of creating a temporary file you should directly store contacts informations in a variable.